backend/core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java

```diff
@@ -230,57 +252,74 @@ public void onConnectionCredentialsChanged(ConnectionCredentialsChangedEvent eve
     LOG.debug("Synchronizing connection '{}' after credentials changed", connectionId);
     var bindingsForUpdatedConnection = configurationRepository.getBoundScopesToConnection(connectionId);
     // Clear the synchronization timestamp for all the scopes so that sync is not skipped
-    bindingsForUpdatedConnection.forEach(boundScope -> synchronizationTimestampRepository.clearLastSynchronizationTimestamp(boundScope.getConfigScopeId()));
+    bindingsForUpdatedConnection.forEach(boundScope -> {
+      scopeSynchronizationTimestampRepository.clearLastSynchronizationTimestamp(boundScope.getConfigScopeId());
+      bindingSynchronizationTimestampRepository.clearLastSynchronizationTimestamp(new Binding(connectionId, boundScope.getSonarProjectKey()));
+    });
     synchronizeConnectionAndProjectsIfNeededAsync(connectionId, bindingsForUpdatedConnection);
   }
 
   private void synchronizeConnectionAndProjectsIfNeededAsync(String connectionId, Collection<BoundScope> boundScopes) {
     var cancelMonitor = new SonarLintCancelMonitor();
     cancelMonitor.watchForShutdown(scheduledSynchronizer);
-    scheduledSynchronizer.submit(() ->
-      serverApiProvider.getServerApi(connectionId)
-        .ifPresent(serverApi -> synchronizeConnectionAndProjectsIfNeededSync(connectionId, serverApi, boundScopes, cancelMonitor)));
+    scheduledSynchronizer.submit(() -> serverApiProvider.getServerApi(connectionId)
+      .ifPresent(serverApi -> synchronizeConnectionAndProjectsIfNeededSync(connectionId, serverApi, boundScopes, cancelMonitor)));
   }
 
   private void synchronizeConnectionAndProjectsIfNeededSync(String connectionId, ServerApi serverApi, Collection<BoundScope> boundScopes, SonarLintCancelMonitor cancelMonitor) {
```

Comment:
`synchronizeConnectionAndProjectsIfNeededSync` method seems a little bit too complex to me. It would be nice if you could extract some parts into new method(s).

Reply:
I agree. I created this [ticket](https://sonarsource.atlassian.net/browse/SLCORE-761) to share what we discussed in the past about this synchronization process, so I think we can tackle the complexity as part of this ticket